Add missing Restore option for ArtifactsPath and tests

baronfel opened the pull request:
Fixes #41530

The commands that implicitly drive MSBuild by forwarding properties should all be available across the set of MSBuild-invoking commands. An important exception to this rule is the `dotnet msbuild` command itself, which is a kind of `low-level` command that doesn't apply as many convenience tools.

baronfel committed:
add missing option + tests

nagilson reviewed (approved):
This looks good to me, I like the way you designed the tests to verify each of the options are on each of the build related commands, very clean :)

baronfel commented:
> I like the way you designed the tests to verify each of the options are on each of the build related commands, very clean :)

Yeah, I fully expect to want to establish more baselines for options (right off the top of my head I can see use for `-c`, `-r`, `-f`, etc) and I wanted to make sure that was super easy to do.

baronfel committed:
make the artifacts path option part of the explicit restore, not the implicit restore

without this, the option is applied multiple times to implicit restores triggered by build, test, etc
